dist2src/worker/processor.py

```diff
@@ -1,13 +1,79 @@
 # Copyright Contributors to the Packit project.
 # SPDX-License-Identifier: MIT
 
+import os
 from logging import getLogger
-from typing import Optional
+from pathlib import Path
+import git
+
+from ogr import PagureService
+from dist2src.core import Dist2Src
 
 logger = getLogger(__name__)
 
 
 class Processor:
-    def process_message(self, event: dict, **kwargs) -> Optional[dict]:
+    def process_message(cls, event: dict, **kwargs):
+        workdir = Path(os.getenv("D2S_WORKDIR", "/dist2src"))
+        dist_git_url = os.getenv("D2S_DIST_GIT_URL", "https://git.centos.org")
+        src_git_url = os.getenv("D2S_SRC_GIT_URL", "https://git.stg.centos.org")
+        src_git_token = os.getenv("D2S_SRC_GIT_TOKEN")
+        dist_git_namespace = os.getenv("D2S_DIST_GIT_NAMESPACE", "rpms")
+        src_git_namespace = os.getenv("D2S_SRC_GIT_NAMESPACE", "source-git")
+        fullname = event["repo"]["fullname"]
+        name = event["repo"]["name"]
+
         logger.info(f"Processing message with {event}")
-        return None
+        # is this a repository in the rpms namespace?
+        if not fullname.startswith(dist_git_namespace):
+            logger.info(
+                f"Ignore update event for {fullname}. Not in the '{dist_git_namespace}' namespace."
+            )
+            return
+        # Which branch was updated?
+        branches_watched = os.getenv("D2S_BRANCHES_WATCHED", "c8s,c8").split(",")
+        branch = event["branch"]
+
+        if event["branch"] not in branches_watched:
+            logger.info(
+                f"Ignore update event for {fullname}. "
+                "Branch {event['branch']} is not one of the watched branches: {branches_watched}."
+            )
+            return
+        # Does this repository have a source-git equvalent?
+        service = PagureService(instance_url=src_git_url, token=src_git_token)
+        project = service.get_project(namespace=src_git_namespace, repo=name)
+        if not project.exists():
+            # Log something, maybe count this
+            logger.info(
+                f"Ignore update event for {fullname}. "
+                "The corresponding source-git repo does not exist."
+            )
+            return
+        # clone repo from rpms/ and checkout the branch
+        dist_git_repo = git.Repo.clone_from(
+            f"{dist_git_url}/{fullname}.git", f"{workdir}/{fullname}", branch=branch
+        )
+        # make sure the commit is the one we are expecting
+        if dist_git_repo.branches[branch].commit.hexsha != event["end_commit"]:
+            raise RuntimeError(
+                f"HEAD of {branch} is not matching {event['end_commit']}, as expected."
+            )
```

Comment:
is this really an error? I'd say it's a warning - to me, the content of a dist-git repo is more important than an event, which is either incorrect or a race condition - hence I'd update here and just warn this situation happened